src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -325,9 +325,9 @@ public virtual void WriteTo(TextWriter writer)
         /// <exception cref="ArgumentException"><paramref name="checksumAlgorithm"/> is not supported.</exception>
         public SourceText GetText(Encoding? encoding = null, SourceHashAlgorithm checksumAlgorithm = SourceHashAlgorithm.Sha1)
         {
-            var builder = new StringBuilder(this.Green.FullWidth);
-            this.WriteTo(new StringWriter(builder));
-            return new StringBuilderText(builder, encoding, checksumAlgorithm);
+            var writer = SourceTextWriter.Create(encoding, checksumAlgorithm, this.Green.FullWidth);
```

Comment:
this helper gets an appropriate writer depending on the length of the final source text. For <85k (the loh limit) you get a simple string writer. For >85k you get a chunking writer.

Reply:
Are you measuring the LOH allocations on .NET Framework or Core? On Core I would be surprised that we still got the LOH allocations. Internally `StringBuilder` should be chucking to avoid the LOH. 

Obvisouly still a great win but want to make sure this all lines up with my mental models here.

Reply:
@jaredpar  this is all for inproc (the place where we have to take nodes and apply them back to the text that teh vs workspace cares about).

I have not seen this show up in OOP (likely for this reason).

Another reason to move all the things to .net core! :)